tests/test_fastapi_roles.py

```diff
@@ -0,0 +1,209 @@
+from typing import Awaitable, Callable, Tuple
+
+import pytest
+from fastapi import FastAPI, HTTPException, Request, Response
+from fastapi.testclient import TestClient
+
+# For tracing test
+from opentelemetry.sdk import trace as tracesdk
+
+# from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import SimpleSpanProcessor
+from opentelemetry.sdk.trace.export.in_memory_span_exporter import InMemorySpanExporter
+
+# Public API
+from dbos import DBOS, DBOSContextEnsure
+
+# Private API because this is a unit test
+from dbos.context import assert_current_dbos_context
+from dbos.error import DBOSDuplicateWorkflowEventError, DBOSNotAuthorizedError
+from dbos.system_database import GetWorkflowsInput
+from dbos.tracer import dbos_tracer
+
+
+@pytest.mark.order(1)
+def test_simple_endpoint(dbos_fastapi: Tuple[DBOS, FastAPI]) -> None:
+    # Set up a simple in-memory span exporter for testing
+    exporter = InMemorySpanExporter()
+    span_processor = SimpleSpanProcessor(exporter)
+    provider = tracesdk.TracerProvider()
+    provider.add_span_processor(span_processor)
+    dbos_tracer.set_provider(provider)
+
+    dbos, app = dbos_fastapi
+
+    @app.middleware("http")
+    async def authMiddleware(
+        request: Request, call_next: Callable[[Request], Awaitable[Response]]
+    ) -> Response:
+        with DBOSContextEnsure() as ctx:
+            ctx.set_authentication("user1", ["user", "engineer"])
+            try:
+                response = await call_next(request)
+                return response
+            finally:
+                ctx.set_authentication(None, None)
```

Comment:
Why is this needed?  Is it just for tests?

Reply:
It may be better to restore what was there before.  But it seems like good practice to undo the change made for this call scope rather than leaving state around where who knows what could happen.

Reply:
Changed to restore previous